src/Orleans.Runtime.Abstractions/Silo/SiloOptions.cs

```diff
@@ -25,4 +27,28 @@ public class SiloOptions
         public bool FastKillOnCancelKeyPress { get; set; } = DEFAULT_FAST_KILL_ON_CANCEL;
```

Comment:
not sure when this was added, but I wouldn't put this in Orleans' specific SiloOptions. We didn't copy everything related to IHost but the IHostLifetime already has things such as these when you use the ConsoleHostLifetime way of hosting... Not suggesting you copy everything related to that (or maybe you do want to), but at least keep it isolated, since it will go away when using Microsoft.Extensions.Hosting

Reply:
@jdom it should go away, but we're not using Microsoft.Extensions.Hosting yet. I see that we were both against having any kind of Ctrl+C handling in the first place: https://github.com/dotnet/orleans/pull/3109

Hopefully we can remove it soon.

Reply:
Added to https://github.com/dotnet/orleans/issues/3957